### PR TITLE
Add fatal signal backtrace

### PR DESCRIPTION
**Pull request by bazsi**
This adds handlers for SIGSEGV and SIGABRT at least on x86_64/arm based linux systems and then reports a crash report in case these signals are triggered.

### CODE REVIEW

**Review by bshifter — DISMISSED** [review]
lgtm

**Comment by bazsi** [comment]
This now only depends on the 4 commits in #357

**Comment by bazsi** [comment]
musl lacks backtrace() and backtrace_symbols() functions. There are replacement libraries (libexecinfo and libbacktrace) but libexecinfo was archived and removed from Alpine. Libbacktrace is not even packaged.

With that this branch is now on hold.

**Comment by bazsi** [comment]
There's now a new iteration that should also work on Alpine Linux and uses libunwind to produce the backtrace.

File: cmake/syslog-ng-config.h.in [hunk 1]
```diff
@@ -106,3 +106,4 @@
 #cmakedefine01 SYSLOG_NG_HAVE_SO_MEMINFO
 #cmakedefine01 SYSLOG_NG_ENABLE_AFSOCKET_MEMINFO_METRICS
 #cmakedefine01 SYSLOG_NG_HAVE_IV_WORK_POOL_SUBMIT_CONTINUATION
+#cmakedefine01 SYSLOG_NG_ENABLE_LIBUNWIND
```
Comment (MrAnno) [inline cmake/syslog-ng-config.h.in:109]: This needs to be set somewhere based on libunwind check results.

**Review by MrAnno — COMMENTED** [review]
LGTM otherwise.

File: cmake/syslog-ng-config.h.in (hunk 1 above)
Comment (MrAnno) [reply cmake/syslog-ng-config.h.in:109]: Meh, I think we let this CMake idea go a long time ago (in production). It is extremely useful during development as it reduces compilation and configuration time to a few seconds. If we used it the modern way, it would also be much easier to use it.

TLDR: I'm merging this for v4.10.0. Let's fix the CMake part separately when we have time.

**Review by MrAnno — DISMISSED** [review]

**Review by MrAnno — APPROVED** [review]
